config-model/src/main/java/com/yahoo/vespa/model/application/validation/BundleValidator.java

```diff
@@ -66,13 +74,71 @@ public void validateOSGIHeaders(DeployLogger deployLogger, JarFile jarFile) thro
         for (String header : requiredOSGIHeaders) {
             if (!mfAttributes.contains(header)) {
                 throw new IllegalArgumentException("Required OSGI header '" + header +
-                        "' was not found in manifest in '" + jarFile.getName() + "'");
+                        "' was not found in manifest in '" + jarPath + "'");
             }
         }
 
         if (attributes.getValue("Bundle-Version").endsWith(".SNAPSHOT")) {
-            deployLogger.logApplicationPackage(Level.WARNING, "Deploying snapshot bundle " + jarFile.getName() +
+            deployLogger.logApplicationPackage(Level.WARNING, "Deploying snapshot bundle " + jarPath +
                     ".\nTo use this bundle, you must include the qualifier 'SNAPSHOT' in  the version specification in services.xml.");
         }
+
+        if (attributes.getValue("Import-Package") != null) {
+            validateImportedPackages(deployLogger, jarPath, mf);
+        }
+    }
+
+    private static void validateImportedPackages(DeployLogger deployLogger, String jarPath, Manifest manifest) {
+        Domain osgiHeaders = Domain.domain(manifest);
+        Parameters importPackage = osgiHeaders.getImportPackage();
+        Map<DeprecatedArtifact, List<String>> deprecatedPackagesInUse = new HashMap<>();
+
+        importPackage.forEach((packageName, attrs) -> {
+            VersionRange versionRange = attrs.getVersion() != null
+                    ? VersionRange.parseOSGiVersionRange(attrs.getVersion())
+                    : null;
+
+            for (DeprecatedArtifact deprecatedArtifact : DeprecatedArtifact.values()) {
+                if (deprecatedArtifact.javaPackages.contains(packageName)
+                        && (versionRange == null || deprecatedArtifact.versionDiscriminator.test(versionRange))) {
+                    deprecatedPackagesInUse.computeIfAbsent(deprecatedArtifact, __ -> new ArrayList<>())
+                            .add(packageName);
+                }
+            }
+        });
+
+        deprecatedPackagesInUse.forEach((artifact, packagesInUse) -> {
+            deployLogger.logApplicationPackage(Level.WARNING,
+                    String.format("For JAR file '%s': \n" +
+                            "Manifest imports the following Java packages from '%s': %s. \n" +
+                            "%s",
+                            jarPath, artifact.name, packagesInUse, artifact.description));
+        });
+    }
+
+    private enum DeprecatedArtifact {
+        ORG_JSON("org.json:json",
```

Comment:
I suggest to add a dummy entry with impossible package/artifact name, and use that in the test, to avoid semicolon hack (for empty enum) and a test that expires as soon as we are on the 8 branch.

Reply:
I'll do that in upcoming PR